File: ruby-gem/lib/models/stock.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Stock
require_relative '../core/semantic_object'


module DfcLinkmlConnector
  module Models
    class Stock < Core::SemanticObject
      SEMANTIC_TYPE = "dfc-b:Stock".freeze

      # @return [String]
      attr_accessor :availability_date

      # @return [Float]
      attr_accessor :quantity

      # @return [String]
      attr_accessor :transported_by

      # @return [String]
      attr_accessor :date

      # @return [String]
      attr_accessor :description

      # @return [String]
      attr_accessor :name

      # @return [String]
      attr_accessor :characteristic_of

      # @return [String]
      attr_accessor :dimension

      # @param semanticId [String]
      # @param availabilityDate: nil, quantity: nil, transportedBy: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil
      def initialize(semanticId, availabilityDate: nil, quantity: nil, transportedBy: nil, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil)
        super(semanticId)
        @availability_date = availabilityDate
        @quantity = quantity
        @transported_by = transportedBy
        @date = date
        @description = description
        @name = name
        @characteristic_of = characteristicOf
        @dimension = dimension
        self.semanticType = "dfc-b:Stock"
        registerSemanticProperty("dfc-b:availabilityDate", &method("availability_date")).valueSetter = method("availability_date=")
        registerSemanticProperty("dfc-b:quantity", &method("quantity")).valueSetter = method("quantity=")
        registerSemanticProperty("dfc-b:transportedBy", &method("transported_by")).valueSetter = method("transported_by=")
        registerSemanticProperty("dfc-b:date", &method("date")).valueSetter = method("date=")
        registerSemanticProperty("dfc-b:description", &method("description")).valueSetter = method("description=")
        registerSemanticProperty("dfc-b:name", &method("name")).valueSetter = method("name=")
        registerSemanticProperty("dfc-b:characteristicOf", &method("characteristic_of")).valueSetter = method("characteristic_of=")
        registerSemanticProperty("dfc-b:hasDimension", &method("dimension")).valueSetter = method("dimension=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end

File: typescript-connector/src/models/Stock.ts
// Class from DFC Business Ontology: #Stock
import { SemanticObject } from "../core/SemanticObject.js";
import type { QuantitativeValue } from "./QuantitativeValue.js";

export interface StockParams {
  availabilityDate?: string;
  quantity?: number;
  transportedBy?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
  hasQuantity?: QuantitativeValue;
}
export class Stock extends SemanticObject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Stock";
  }

  availabilityDate?: string;
  quantity?: number;
  transportedBy?: string;
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
  hasQuantity?: QuantitativeValue;

  constructor(
    semanticId: string,
    params?: StockParams,
  ) {
    super(semanticId);
    this.availabilityDate = params?.availabilityDate;
    this.quantity = params?.quantity;
    this.transportedBy = params?.transportedBy;
    this.date = params?.date;
    this.description = params?.description;
    this.name = params?.name;
    this.characteristicOf = params?.characteristicOf;
    this.hasDimension = params?.hasDimension;
    this.hasQuantity = params?.hasQuantity;
    this.semanticType = Stock.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:availabilityDate", () => this.availabilityDate);
    this.registerSemanticProperty("dfc-b:quantity", () => this.quantity);
    this.registerSemanticProperty("dfc-b:transportedBy", () => this.transportedBy);
    this.registerSemanticProperty("dfc-b:date", () => this.date);
    this.registerSemanticProperty("dfc-b:description", () => this.description);
    this.registerSemanticProperty("dfc-b:name", () => this.name);
    this.registerSemanticProperty("dfc-b:characteristicOf", () => this.characteristicOf);
    this.registerSemanticProperty("dfc-b:hasDimension", () => this.hasDimension);
    this.registerSemanticProperty("dfc-b:hasQuantity", () => this.hasQuantity);
  }
  static {
    SemanticObject.typeRegistry.set(Stock.SEMANTIC_TYPE, Stock);
  }
}
